File: Yow.CoD.Finance.Domain/Contracts/PaymentTakenEvent.cs
using System;

namespace Yow.CoD.Finance.Domain.Contracts
{
    public sealed class PaymentTakenEvent : Event
    {
        public PaymentTakenEvent(string transactionId,DateTimeOffset transactionDate, decimal amount)
        {
            TransactionId = transactionId;
            TransactionDate = transactionDate;
            Amount = amount;
        }

        public string TransactionId { get;  }
        public DateTimeOffset TransactionDate { get; }
        public decimal Amount { get; }
    }
}

File: java/domain/src/main/java/com/leecampbell/cod/domain/contracts/PaymentTakenEvent.java
package com.leecampbell.cod.domain.contracts;

import java.math.BigDecimal;
import java.time.OffsetDateTime;

public final class PaymentTakenEvent extends DomainEvent {
    
    private final String transactionId;
    private final OffsetDateTime transactionDateTime;
    private final BigDecimal amount;

    public PaymentTakenEvent(String transactionId, OffsetDateTime transactionDateTime, BigDecimal amount) {
        this.transactionId = transactionId;
        this.transactionDateTime = transactionDateTime;
        this.amount = amount;
    }

    public String transactionId() {
        return this.transactionId;
    }

    public OffsetDateTime transactionDateTime() {
        return this.transactionDateTime;
    }

    public BigDecimal amount() {
        return this.amount;
    }
}